File: cf_sdk/src/main/java/com/example/cf_sdk/logic/changebanksdk/source/IngoDatasource.java
package com.example.cf_sdk.logic.changebanksdk.source;

import com.example.cf_sdk.logic.changebanksdk.parameter.StringParameters;
import com.example.cf_sdk.logic.changebanksdk.parameter.ingo.VerifyIngoSSNParameters;
import com.example.cf_sdk.logic.changebanksdk.response.IngoResponse;

import io.reactivex.Single;

/**
 *
 * Datasource for Ingo API.
 */

public interface IngoDatasource {

    Single<IngoResponse> verifyIngoSSSN(VerifyIngoSSNParameters verifyIngoSSNParameters);

    Single<IngoResponse> getIngoSSOToken(StringParameters stringParameters);

}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/source/datasource/IngoDatasource.kt
package com.example.cf_sdk.changebankapi.source.datasource

import com.example.cf_sdk.changebankapi.parameter.StringParameters
import com.example.cf_sdk.changebankapi.parameter.ingo.VerifyIngoSSNParameters
import com.example.cf_sdk.changebankapi.response.IngoResponse
import io.reactivex.Single

/**
 *
 *
 * Datasource for Ingo API.
 */
interface IngoDatasource {
    fun verifyIngoSSSN(verifyIngoSSNParameters: VerifyIngoSSNParameters?): Single<IngoResponse?>?
    fun getIngoSSOToken(stringParameters: StringParameters?): Single<IngoResponse?>?
}